Gems/Atom/Tools/ShaderManagementConsole/Code/Source/Window/ShaderManagementConsoleTableView.cpp

```diff
@@ -99,36 +160,89 @@ namespace ShaderManagementConsole
             m_shaderOptionDescriptors.push_back(shaderOptionDescriptor);
         }
 
-        // Sort all of the descriptors by name so that the columns are arranged in alphabetical order
-        AZStd::sort(
-            m_shaderOptionDescriptors.begin(),
-            m_shaderOptionDescriptors.end(),
-            [](const auto& a, const auto& b)
+        switch (m_columnSortMode)
+        {
+        case Alpha:
             {
-                return a.GetName().GetStringView() < b.GetName().GetStringView();
-            });
+                // Sort descriptors by name
+                AZStd::sort(
+                    m_shaderOptionDescriptors.begin(),
+                    m_shaderOptionDescriptors.end(),
+                    [](const auto& a, const auto& b)
+                    {   //                      small first ('a' first)
+                        return a.GetName().GetStringView() < b.GetName().GetStringView();
+                    });
+            }
+            break;
+        case Rank:
+            {
+                // Sort by shader declaration order
+                AZStd::sort(
+                    m_shaderOptionDescriptors.begin(),
+                    m_shaderOptionDescriptors.end(),
+                    [](const auto& a, const auto& b)
+                    {   //  small first (order 0 is highest)
+                        return a.GetOrder() < b.GetOrder();
+                    });
+            }
+            break;
+        case Cost:
+            {
+                // Sort by cost (high score first)
+                AZStd::sort(
+                    m_shaderOptionDescriptors.begin(),
+                    m_shaderOptionDescriptors.end(),
+                    [](const auto& a, const auto& b)
+                    {   //      big first (high score is more important)
+                        return a.GetCostEstimate() > b.GetCostEstimate();
+                    });
+            }
+            break;
+        }
 
         // Fill in the header of each column with the descriptor name
-        for (int column = 0; column < columnCount(); ++column)
+        for (int column = 1; column < columnCount(); ++column)
         {
-            const auto& shaderOptionDescriptor = m_shaderOptionDescriptors[column];
-            setHorizontalHeaderItem(column, new QTableWidgetItem(shaderOptionDescriptor.GetName().GetCStr()));
+            const auto& shaderOptionDescriptor = m_shaderOptionDescriptors[column - 1];
+            auto* tableItem = new QTableWidgetItem(shaderOptionDescriptor.GetName().GetCStr());
+            tableItem->setToolTip(tr("cost %1").arg(shaderOptionDescriptor.GetCostEstimate()));
+            setHorizontalHeaderItem(column, tableItem);
         }
+        setHorizontalHeaderItem(0, new QTableWidgetItem(""));
+
+        QIcon emptyOptionIcon("D:/emptyoption.png");
```

Comment:
I have no idea of the resource submission proper process. Do we need to prepare multi DPI images, or can I just push that png somewhere in the repository, is there a impact on installation (I mean deployment of the app to user machine) etc..

Reply:
There is an icon folder under Source/Window. Put your files there and update the QRC in the folder above. Regarding DPI, UI has moved most images away from PNG to SVG, scalable vector graphics, files.